azure-pipelines/network-automation/e2e-network-tests.yml

```diff
@@ -0,0 +1,194 @@
+name: $(Build.BuildId)_$(Build.DefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger: none
+pr: none
+
+variables:
+  engineeringProjectId: 'fac9d424-53d2-45c0-91b5-ef6ba7a6bf26'
+  brokerHostPipelineId: 1432
+  azureSamplePipelineId: 1458
+  app: msalAutomationApp.apk
+  testApp: msalAutomationAppAndroidTest.apk
+  deviceId: flame
+  deviceAndroidVersion: 29
+  firebaseTimeout: 45m
+  resultsHistoryName: "Network test"
+  resultsDir: "network-test-$(Build.BuildId)-$(Build.BuildNumber)"
+
+pool:
+  vmImage: ubuntu-latest
+
+
+jobs:
+  - job: one
+    displayName: E2E Network test run
+    timeoutInMinutes: 60
+    steps:
+      # Setup
+      - checkout: self
+        clean: true
+        submodules: recursive
+        persistCredentials: True
+      - task: AzureKeyVault@2
+        displayName: 'Get Key vault AndroidAutomationRunnerAppSecret'
+        inputs:
+          azureSubscription: 'MSIDLABS_ANDROID_KV'
+          KeyVaultName: 'ADALTestInfo'
+          SecretsFilter: 'AndroidAutomationRunnerAppSecret'
+          RunAsPreJob: false
+      - task: DownloadSecureFile@1
+        displayName: 'Download Firebase Service Account Key File'
+        name: gcServiceAccountKey
+        inputs:
+          secureFile: AndroidFirebaseServiceAccountKey.json
+          retryCount: 5
+      - task: CmdLine@1
+        displayName: 'Set MVN Access Token in Environment'
+        inputs:
+          filename: echo
+          arguments: '##vso[task.setvariable variable=ENV_VSTS_MVN_ANDROIDADACCOUNTS_ACCESSTOKEN]$(mvnAccessToken)'
+      # Assemble MSAL Automation apks
+      - task: Gradle@1
+        displayName: 'Assemble MSAL Automation App'
+        inputs:
+          tasks: 'clean msalautomationapp:assembleLocalBrokerMicrosoftAuthenticator -PlabSecret=$(AndroidAutomationRunnerAppSecret)'
+          publishJUnitResults: false
+      - task: Gradle@1
+        displayName: 'Assemble MSAL Automation App Instrumented Tests'
+        inputs:
+          tasks: 'msalautomationapp:assembleLocalBrokerMicrosoftAuthenticatorDebugAndroidTest -PlabSecret=$(AndroidAutomationRunnerAppSecret)'
+          publishJUnitResults: false
+      - script: |
+          baseApkPath="$(Build.SourcesDirectory)/msalautomationapp/build/outputs/apk"
+          appApk="$baseApkPath/localBrokerMicrosoftAuthenticator/debug/*.apk"
+          testApk="$baseApkPath/androidTest/localBrokerMicrosoftAuthenticator/debug/*.apk"
+          mv $appApk $(app)
+          mv $testApk $(testApp)
+          ls -lh *.apk
+        displayName: 'Rename APKs'
+        workingDirectory: '$(Build.StagingDirectory)'
+      # Firebase tasks
+      - task: Bash@3
+        displayName: Configure gcloud
+        inputs:
+          targettype: inline
+          script: |
+            gcloud auth activate-service-account --key-file "$(gcServiceAccountKey.secureFilePath)"
+            gcloud config set project $(gCloudProjectId)
+      - task: Bash@3
```

Comment:
I think here is a great opportunity to use matrix strategy.
https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/jobs-job-strategy?view=azure-pipelines